### DISCUSSION TITLE
Deployment urls with seo

### DISCUSSION THREAD

**Discussion by MejanH** [vercel/vercel#5038] [Help] [answered]
I am worrying that having too many random deployment domains is going to affect SEO or not? I can only redirect the domain added in the domain list but there is no option to redirect deployment URLs also.

Sorry, if I am worrying too much.

**Comment by styfle** [comment, marked as answer]
Deployments that are not assigned a Production Domain will be served with the `x-robots-tag: noindex` header. This will prevent search engines from indexing that URL so there is nothing to worry about.

Read more about `x-robots-tag` here: https://vercel.com/docs/edge-network/headers

**Reply by wadamek65** [reply]
What about production deployments that are assigned multiple domains? As I checked they do not have the `noindex` header. Is there any way to fix this or remove these extra unnecessary domains?

**Reply by styfle** [reply]
@wadamek65 Do you have a [custom domain configured](https://vercel.com/docs/custom-domains) in your project settings?

If so, can you share a deployment URL that is not working as expected?

**Reply by wadamek65** [reply]
@styfle Yes and it is working properly. But when I create a production deployment, apart from my own custom domain, three other aliases get assigned automatically. The way I handle SEO is generate the head during the build time by fetching the data from my CMS. So whenever I do a production deployment I end up with the same version of my application on 4 different domains, all with `index,follow` in `meta`.

As far as I know there isn't a way to get rid of the aliases (at least not all of them - I tried manually removing them and managed to delete all but one) so this is quite problematic for me.

Would setting the canonical URL solve all problems regarding indexing?

![image](https://user-images.githubusercontent.com/36732892/112225526-3bd27f80-8c2d-11eb-95d5-ab022cd90d1e.png)

**Reply by styfle** [reply]
@wadamek65 Your production domain is `subscribo.pl`. The others are generated by our system and will automatically add a [`x-robots-tag: noindex`](https://developers.google.com/search/docs/advanced/robots/robots_meta_tag) header. This means that search engines won't index those system-generated domains.

For example, compare the headers between these two:

```
curl -sI https://subscribo.pl
curl -sI https://subscribo-o5nfybmp8-subscribo-org.vercel.app
```

That being said, its never a bad idea to set the [canonical URL](https://developers.google.com/search/docs/advanced/crawling/consolidate-duplicate-urls) manually in your application.

**Reply by wadamek65** [reply]
@styfle Thanks for the explanation - I didn't check the headers returned as I thought they are only used for preview deployments, not production aliases. 

I couldn't find any definitive information whether the header takes precedence over the html tags but common sense would say that they do since they would get parsed first.

I suppose this isn't a problem for me then, thank you! 😁
